src/sdc11073/xmlparsing.py

```diff
@@ -67,7 +69,23 @@ def copy_node(node: etree_._Element, method=copy.deepcopy) -> etree_._Element:
     :param method: method that copies an etree element
     :return: new report node
     """
+    # walk from target to root
+    current = node
+    ns_map_list: typing.List[typing.Dict[str, str]] = []  # saves all namespaces
+    while current is not None:
+        ns_map_list.append({k: v for k, v in current.nsmap.items() if k})  # filter for default namespace
+        current = current.getparent()
+
+    # create new instance
     root_tree = node.getroottree()
-    new_report = method(root_tree.getroot())
-    ns_map = {k: v for k, v in node.nsmap.items() if k}  # filter for default namespace
-    return new_report.xpath(root_tree.getpath(node), namespaces=ns_map)[0]
+    current = method(root_tree.getroot())
+    x_path_steps = root_tree.getpath(node).split('/')[1:]
+    assert len(x_path_steps) == len(ns_map_list)
+
+    # walk from root to target
+    ns_map_list.reverse()
+    for i, step in enumerate(x_path_steps):
+        if i == 0:
+            step = f'/{step}'
+        current = current.xpath(step, namespaces=ns_map_list[i])[0]
+    return current
```

Comment:
I suggest a much simpler solution, which is also 2... 3 times faster:

def copy_node(node: etree_.ElementBase, method=copy.deepcopy) -> etree_.ElementBase:
    new_node = etree_.Element(node.tag,
                              attrib=dict(node.attrib),
                              nsmap=dict(node.nsmap))
    new_node.text = copy.copy(node.text)
    new_node.tail = copy.copy(node.tail)
    new_node.extend((method(child) for child in node))
    return new_node

Reply:
I introduced it as a second copy method and used it where I think the parent is not needed